### ISSUE TITLE
Crash when targeting with the castle guards (hero with Artillery garrisoned)

### ISSUE THREAD

**Issue by 6b6279** [vcmi/vcmi#3872]
**Describe the bug**
Game crashes when I hover over an enemy unit during a town siege with a hero that has Artillery as a secondary skill

**Game logs**
```
Opening BattleAI
Loaded Battle AI
Error: accessing unavailable frame 0:2 in CAnimation!
Error: accessing unavailable frame 0:2 in CAnimation!
Cannot answer the query -1!
Opening BattleAI
Loaded Battle AI
Call to loadBitmap with void fname!
Error: failed to load image 
Animation  failed to load
GetImage error: Request for frame not present in CAnimation! File name: SPRITES/, Group: 0, Frame: 0
/usr/include/c++/13.2.0/array:202: constexpr std::array<_Tp, _Nm>::value_type& std::array<_Tp, _Nm>::operator[](size_type) [with _Tp = unsigned int; long unsigned int _Nm = 187; reference = unsigned int&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Network error receiving a pack. Connection has been closed
```

**To Reproduce**
1. Garrison a hero with Artillery in a town with a Castle or Citadel.
2. Wait until the town is sieged.
3. Upon the start of the combat, hover over an enemy unit when it's one of the tower shooters' turns.
4. Game crashes.

**Expected behavior**
The game shouldn't crash.

**Actual behavior**
The game crashes.

**Did it work earlier?**
Looks like a resurgence of #2938

**Screenshots**
See attached video in #2938

**Version**
 - OS: Arch Linux x86
 - Version: 1.4.5, Flatpak

**Additional context**
Came across the bug whilst playing the second scenario of New Beginning (SoD).

**Comment by 6b6279** [comment]
The assertion seems to be one of those that can be enabled optionally while compiling libc++ (with `_GLIBCXX_ASSERTIONS`, which is defined in, among others, Arch Linux). #3452 seems to address this issue, but it has only been merged into the beta branch.

**Comment by IvanSavenko** [comment]
Yeah, should be fixed in 1.5 release. I do remember fixing this particular scenario.

**Comment by 6b6279** [comment]
Alright, I'll switch to vcmi-git until 1.5 is released. Thanks!
